src/Sarif/Writers/SarifLogger.cs

```diff
@@ -155,52 +154,17 @@ public class SarifLogger : BaseLogger, IDisposable, IAnalysisLogger
             OptionallyEmittedData dataToRemove,
             IEnumerable<string> invocationTokensToRedact,
             IEnumerable<string> invocationPropertiesToLog,
-            string defaultFileEncoding = null,
-            IDictionary<string, HashData> filePathToHashDataMap = null)
+            string defaultFileEncoding = null)
         {
             _run.Invocations ??= new List<Invocation>();
             if (defaultFileEncoding != null)
             {
                 _run.DefaultEncoding = defaultFileEncoding;
             }
 
-            Encoding encoding = SarifUtilities.GetEncodingFromName(_run.DefaultEncoding);
-
             if (analysisTargets != null)
             {
                 _run.Artifacts ??= new List<Artifact>();
-
-                foreach (string target in analysisTargets)
-                {
-                    Uri uri = new Uri(UriHelper.MakeValidUri(target), UriKind.RelativeOrAbsolute);
-
-                    HashData hashData = null;
-                    if (dataToInsert.HasFlag(OptionallyEmittedData.Hashes))
-                    {
-                        filePathToHashDataMap?.TryGetValue(target, out hashData);
-                    }
-
-                    var artifact = Artifact.Create(
-                        new Uri(target, UriKind.RelativeOrAbsolute),
-                        dataToInsert,
-                        encoding,
-                        hashData: hashData);
-
-                    var fileLocation = new ArtifactLocation
-                    {
-                        Uri = uri
-                    };
-
-                    artifact.Location = fileLocation;
-
-                    // This call will insert the file object into run.Files if not already present
-                    artifact.Location.Index = _run.GetFileIndex(
-                        artifact.Location,
-                        addToFilesTableIfNotPresent: true,
-                        dataToInsert: dataToInsert,
-                        encoding: encoding,
-                        hashData: hashData);
-                }
```

Comment:
This code path is used by the single threaded analysis. When hashes is enabled, it was going to generate all hashes and store everything in the SARIF.

The issue is that if we analyze 1k files and we do not produce any results, we would still generate the artifacts.

As an optimization, we will only store the artifact if we have a result.

Reply:
Reviewed all versions that we released and I saw that all versions after 1.4.2 are emitting the artifacts IF the list of analysisTargets is not empty.

Reply:
Looking at the other variables, looks like we tried to improve this behavior adding the _persistArtifacts, but that is occurring too late (after we already added the artifact to the run)